monkit.lock

```diff
@@ -61,3 +61,5 @@ storj.io/storj/satellite/repair/checker."remote_segments_lost" IntVal
 storj.io/storj/satellite/repair/checker."remote_segments_needing_repair" IntVal
 storj.io/storj/satellite/satellitedb."audit_reputation_alpha" FloatVal
 storj.io/storj/satellite/satellitedb."audit_reputation_beta" FloatVal
+storj.io/storj/storagenode/contact."satellite_contact_request" Meter
```

Comment:
Did Data Science request that these metrics be added to the lock list? I don't think we should be adding things here by default.

Reply:
We need to lock those metrics to avoid changes without our consent.

Reply:
I actually think we should be adding things here by default. I just had a brief convo with Patrick about it. The purpose of `monkit.lock` is to allow us to have a dictionary for the stats we are reporting to monkit. That way we don't accidentally remove reporting when code moves around.
He couldn't think about any situation off the top of his head where we'd be reporting a monkit stat from this repository and we *wouldn't* want it in the lock file.

Reply:
One big reason is that each locked metric adds maintenance overhead: in effect, an obstacle in the way of code reorganization or removal of code. The obstacle may not be very large—I expect that in most cases just a heads-up to DS is good enough to allow changes to location or semantics—but there are already over a thousand separate `mon.X()` calls in our codebase and it will really add up.

It seems a lot more effective to come at this with an "opt-in" approach; any metrics that DS is actively using in dashboards/reports/whatever can be added to the lock list, and anything else is left unencumbered.

Finally, if we expect "locked by default" for stats, we should probably add locks to all existing metrics to avoid confusion ("but why aren't all these other ones locked?"). Currently less than 6% of them are locked.

Reply:
Excluding the `mon.Task()` calls that are in basically every function across the codebase (and probably consist of the majority of the `mon.X()` calls you are referring to), my understanding after talking with Patrick is that any specific stat that we added with intent should be in the lock file. We should eventually be adding locks to the existing metrics that aren't currently locked. I added some earlier today to some audit metrics that were added before `monkit.lock` existed. 
What datascience is concerned about is that when the code inevitably changes, that metrics that we added intentionally will go away, either because they were not kept track of properly, or because code was refactored in a way that made the metric obsolete. So if we are deliberately adding a metric (`mon.IntVal`, `mon.FloatVal`, `mon.Counter`, `mon.Meter`, etc...), we should make sure to lock it.

Reply:
rafael is the DS team, so this PR counts as a valid request.

Reply:
> rafael is the DS team, so this PR counts as a valid request.

Heh, ok, I forgot that part :D

Reply:
> Excluding the mon.Task() calls that are in basically every function across the codebase (and probably consist of the majority of the mon.X() calls you are referring to), my understanding after talking with Patrick is that any specific stat that we added with intent should be in the lock file. We should eventually be adding locks to the existing metrics that aren't currently locked. I added some earlier today to some audit metrics that were added before monkit.lock existed.

Ah. Okay, if `mon.Task` is not included, that seems a lot more reasonable. If we want this to be a new rule, though, can we please (a) announce it, (b) add it to our [code style](https://github.com/storj/storj/wiki/Style) doc, and (c) make a quick linter to enforce? Otherwise it will end up being inconsistently applied, which seems potentially even worse than not having the rule at all (because there will be an expectation that stats are safe from disappearing or moving by default).

Reply:
Yeah I agree. I think we should also add it to the PR checklist. I can put that and updating the style doc on my todo list for tomorrow